l2gersync/evm_downloader_legacy.go

```diff
@@ -172,21 +200,17 @@ func (d *downloaderFEP) getGERsFromIndex(
 		if err != nil {
 			return nil, fmt.Errorf("failed to retrieve l1 info tree leaf for index=%d: %w", i, err)
 		}
-		gers = append(gers,
-			&GlobalExitRootInfo{
-				L1InfoTreeIndex: i,
-				GlobalExitRoot:  info.GlobalExitRoot,
-			})
+		gers = append(gers, newGlobalExitRootInfo(info.GlobalExitRoot, i, info.BlockNumber, info.BlockPosition))
 	}
 
 	return gers, nil
 }
 
-func (d *downloaderFEP) populateGreatestInjectedGER(b *sync.EVMBlock, gerInfos []*GlobalExitRootInfo) {
+func (d *downloaderLegacy) findLatestInjectedGER(gerInfos []*GlobalExitRootInfo) *GlobalExitRootInfo {
```

Comment:
Are these GERs sorted somehow, by index or block? If not, I don't think our latest injected logic is good.